### ISSUE TITLE
Reproducible label-aware augmentation artifacts in the computer-vision example

### ISSUE THREAD

**Issue by ternaus** [zenml-io/zenml#5191]
# Feature Description

Would you be open to a small AlbumentationsX example that makes a label-aware augmentation policy an explicit, versioned part of a ZenML computer-vision pipeline?

The current [`examples/computer_vision`](https://github.com/zenml-io/zenml/tree/4cd0be89aee0ad0cc0cc2b237c28c3d490729e45/examples/computer_vision) workflow already provides a useful end-to-end path from a FiftyOne COCO export through Ultralytics training, model artifact versioning, inference, and FiftyOne analysis. I would preserve that workflow and add either an optional augmentation variant or a separate small example, whichever is easier for the maintainers to own.

# Problem or Use Case

In a reproducible object-detection pipeline, the augmentation policy and seed are part of the data lineage: changing either can produce a different training dataset even when the source dataset is unchanged.

The current example versions the exported dataset and trained model, but it does not have an explicit augmentation-policy artifact or a label-aware dataset-preparation step between them. A stochastic preprocessing step also needs sample-level randomness that remains stable if sample or worker execution order changes.

# Proposed Solution

The proposed example would keep the implementation bounded to two ZenML steps:

1. Build an AlbumentationsX pipeline and return `A.to_dict(pipeline)` as a normal ZenML `dict` artifact.
2. Reconstruct it with `A.from_dict()`, apply synchronized image and bounding-box transforms to the exported YOLO dataset, and return the resulting directory as a normal ZenML `Path` artifact for the existing trainer.

A simplified core would look like this:

```python
import albumentations as A

policy = A.Compose(
    [
        A.HorizontalFlip(p=0.5),
        A.RandomBrightnessContrast(p=0.2),
    ],
    bbox_params=A.BboxParams(
        coord_format="yolo",
        label_fields=["class_labels"],
    ),
)

policy_artifact = A.to_dict(policy)

# In the downstream dataset-preparation step:
policy = A.from_dict(policy_artifact)
sample = policy(
    image=image,
    bboxes=bboxes,
    class_labels=class_labels,
    invocation_seed=stable_sample_seed,
)
```

The sample seed would be derived from a pipeline parameter and a stable sample identifier, so the generated dataset does not change merely because an orchestrator processes samples in a different order.

The resulting ZenML lineage would connect:

```text
source dataset -> augmentation policy + seed -> transformed dataset -> trained model
```

This would use ZenML's existing materialization for `dict` and `Path`; it would not require a new ZenML integration, stack component, or custom materializer.

# Alternatives Considered

- Leaving augmentation entirely inside the training implementation does not make the policy or transformed dataset a separate lineage artifact.
- A custom materializer for `A.Compose` is possible, but a serialized dictionary is simpler and demonstrates the same workflow with existing ZenML primitives.
- Changing the current computer-vision example by default would be unnecessarily disruptive. An optional step or separate example would preserve the existing path.

# Additional Context

I would be happy to implement the example after the maintainers confirm whether it belongs as an optional branch of `examples/computer_vision` or as a smaller standalone example.

Dependency boundary, stated up front:

- The public `albumentationsx` package is AGPL-3.0-only.
- It imports as `albumentations`.
- Importing it requires PyTorch. PyTorch is intentionally not pinned in its package metadata because users need to select the appropriate CPU, CUDA, or MPS build. The example would document that runtime choice rather than add a universal PyTorch pin.
- AlbumentationsX would remain scoped to this example and would not become a ZenML runtime dependency.

# Priority

Low - Nice to have

# Code of Conduct

- [x] I agree to follow this project's Code of Conduct

**Comment by modelpath-dev** [comment]
I will take this issue. Please assign it to me.

The issue seems to be about adding a label-aware augmentation policy to the existing computer-vision example in ZenML. The current workflow lacks an explicit augmentation-policy artifact. I would first look at the `examples/computer_vision` directory to understand the existing pipeline. Then, I would add a step to build and serialize an AlbumentationsX pipeline as a ZenML artifact. The next step would reconstruct and apply this policy to the dataset. This should ensure reproducibility by maintaining stable sample-level randomness.

**Comment by modelpath-dev** [comment]
Could you please assign this issue to me? I will start by reviewing the existing computer-vision example and then work on integrating the label-aware augmentation policy as discussed. Let me know if there are any specific preferences or guidelines I should follow.
